ui/app/components/secret-engine/configurable-secret-engine-details.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import { service } from '@ember/service';
+import Component from '@glimmer/component';
+import errorMessage from 'vault/utils/error-message';
+import { tracked } from '@glimmer/tracking';
+
+import type Store from '@ember-data/store';
+import type SecretEngineModel from 'vault/models/secret-engine';
+import type AdapterError from '@ember-data/adapter';
+import type Model from '@ember-data/model';
+
+/**
+ * @module ConfigurableSecretEngineDetails
+ * `ConfigurableSecretEngineDetails` is used by configurable secret engines (AWS, SSH) to show either an API error, configuration details, or a prompt to configure the engine. Which of these is shown is determined by the engine type and whether the user has configured the engine yet.
+ *
+ * @example
+ * ```js
+ * <ConfigurableSecretEngineDetails @model={{this.model}} />
```

Comment:
I think it's helpful when this component uses the correct syntax for implementation, so like `SecretEngine::ConfigurableSecretEngineDetails`. Thinking about the name more, since this is already in the `secret-engine` directory I think we can simplify this component name to be  `SecretEngine::ConfigurationPage`.